action/kubeconfig.go

```diff
@@ -114,7 +115,7 @@ func (c *ClusterConfig) writeKubeconfig(clusterInfo *eks.DescribeClusterOutput)
 				Value: c.Alias,
 			},
 		},
-		APIVersion: "client.authentication.k8s.io/v1beta1",
+		APIVersion: "client.authentication.k8s.io/v1",
```

Comment:
this is breaking change not a fix, can you try to get the server kubernetes version and set API Version compatible to Kubernetes Server Version? you should have kubernetes version available with `clusterInfo.Cluster.Version` https://github.com/signavio/aws-mfa-login/blob/bb218ef16174fe434f9f1dabbf4da6c19f8c9647/action/kubeconfig.go#L97

Reply:
> api version v1 is supported since what Kubernetes version?

If I'm not mistaken, v1 has been introduced since v1.6. See [CHANGELOG-1.6.md#authentication](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.6.md#authentication) and beta was deprecated since 1.19, see [CHANGELOG-1.19.md#deprecation](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.19.md#deprecation)

>this is breaking change not a fix, can you try to get the server kubernetes version and set API Version compatible to Kubernetes Server Version? you should have kubernetes version available with clusterInfo.Cluster.Version
Good point. I like it. Will change that.

Reply:
@karlderkaefer I did a lot of research, and we can avoid over-engineering and go with the V1 version.
- I tested clusters with v1, and it did worked.
- `v1` has been supported since v1.6. This means this is supported on all versions of the cluster we have.
- `v1beta1` will be removed in 1.29
- aws-cli has the version `v1beta1` as a fallback, but `v1` is also fine. see https://github.com/aws/aws-cli/blob/cdaaad0595922913d706f47f8efee47bd39ff19f/awscli/customizations/eks/get_token.py#L161-L165
- Additionally, AWS doesn't have any cluster below 1.19. [AWS Cluster Supported Version](https://relnotes.k8s.io/?markdown=authentication.k8s.io)

Conclusion: hardcoded v1 is fine.